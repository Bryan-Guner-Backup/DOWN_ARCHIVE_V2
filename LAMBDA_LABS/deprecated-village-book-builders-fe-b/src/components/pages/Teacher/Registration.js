import React, { useState } from 'react';
import { Redirect, useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import { Form, Input, Button } from 'antd';
import { register } from '../../../state/actions';
import { FormContainer, layout, tailLayout } from '../../common/FormStyle';
// import registerReducer from '../../../state/reducers'

const initialState = {
  first_name: '',
  last_name: '',
  email: '',
  password: '',
};

const Registration = ({ register, registered }) => {
  const [theData, setTheData] = useState(initialState);
  const [form] = Form.useForm();
  let history = useHistory();

  const handleChange = e => {
    setTheData({ ...theData, [e.target.name]: e.target.value });
  };
  const onSubmit = async () => {
    console.log('test');
    await register(theData);
    history.push('/register_pt2');
  };

  return registered ? (
    <Redirect to="/login" />
  ) : (
    <FormContainer>
      <Form onFinish={onSubmit} form={form} {...layout}>
        <Form.Item {...tailLayout}>
          <h1>Register</h1>
        </Form.Item>

        <Form.Item
          label="First_Name"
          name="first_name"
          rules={[{ required: true, message: 'First name is required.' }]}
        >
          <Input
            type="text"
            name="first_name"
            value={theData.first_name}
            onChange={e => handleChange(e)}
          />
        </Form.Item>

        <Form.Item
          label="Last_Name"
          name="last_name"
          rules={[{ required: true, message: 'Last name is required.' }]}
        >
          <Input
            type="text"
            name="email"
            value={theData.last_name}
            onChange={e => handleChange(e)}
          />
        </Form.Item>
        <Form.Item
          label="email"
          name="email"
          rules={[{ required: true, message: 'Email is required.' }]}
        >
          <Input
            type="text"
            name="email"
            value={theData.email}
            onChange={e => handleChange(e)}
          />
        </Form.Item>

        <Form.Item
          label="password"
          name="password"
          rules={[{ required: true, message: 'Password is required.' }]}
        >
          <Input
            type="password"
            name="password"
            value={theData.password}
            onChange={e => handleChange(e)}
          />
        </Form.Item>
        <Form.Item {...tailLayout}>
          <Button htmlType="submit">Register</Button>
        </Form.Item>
      </Form>
      <h2>Register to become a teacher!</h2>
    </FormContainer>
  );
};

const mapStateToProps = state => {
  return {
    registered: state.authReducer.registered,
  };
};

export default connect(mapStateToProps, { register })(Registration);
