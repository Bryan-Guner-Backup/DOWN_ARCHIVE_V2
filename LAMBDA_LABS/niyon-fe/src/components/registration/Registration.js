import React from 'react'
import { useForm } from 'react-hook-form'
import { registerUser } from '../apiStuff/axiosWithAuth'
import Header from '../header/Header'

const Registration = (props) => {
  const { register, handleSubmit, watch, errors } = useForm()
  const onSubmit = (data) => handleOnSubmit(data)

  const handleOnSubmit = (props) => {
    const userDetails = {
      email: props.email,
      user_type: props.user_type,
      password: props.password
    }
    window.localStorage.setItem('user_type', props.user_type)
    registerUser(userDetails)
      .then((res) => {
        console.log('hitting')
        window.localStorage.setItem('token', res.data.token)
        window.localStorage.setItem('id', res.data.user.id)
        window.location = '/profile'
      })
      .catch((err) => {
        console.log('not hitting error')
        console.log(err)
      })
  }

  return (
    <div className="formWrap">
      <Header />
      <form className="formRegister" onSubmit={handleSubmit(onSubmit)}>
        <input
          type="email"
          placeholder="Email"
          name="email"
          ref={register({ required: true })}
        />
        {errors.email && (
          <p style={{ color: 'orange', marginTop: 10 }}>
            Email is required
          </p> /*eslint-disable */
        )}
        <input
          type="password"
          placeholder="Password"
          name="password"
          ref={register({ required: true })}
        />
        {errors.password && (
          <p style={{ color: "orange", marginTop: 10 }}>Password is required</p>
        )}
        <input
          type="password"
          placeholder="Repeat Password"
          name="password2"
          ref={register({
            validate: (value) => {
              return value === watch("password"); // value is from password2 and watch will return value from password1
            },
          })}
        />
        {errors.password2 && (
          <p style={{ color: "orange", marginTop: 10 }}>
            Passwords do not match
          </p>
        )}
        <select name="user_type" ref={register({ required: true })}>
          <option value="Mentor">Mentor</option>
          <option value=" Mentee">Mentee</option>
        </select>
        <button type="submit">Register</button>
      </form>
    </div>
  );
};

export default Registration;
