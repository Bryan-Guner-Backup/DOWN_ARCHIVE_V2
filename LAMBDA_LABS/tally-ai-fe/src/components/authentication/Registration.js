import React, { useState } from 'react'
import axios from 'axios'
import { Link } from 'react-router-dom'
import { Formik, Form } from 'formik'

import { Button, LinearProgress } from '@material-ui/core'
import CssBaseline from '@material-ui/core/CssBaseline'
import TextField from '@material-ui/core/TextField'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import Container from '@material-ui/core/Container'
import GoogleLoginBtn from '../google/GoogleLoginBtn';

import RegistrationSchema from './RegistrationSchema'

const useStyles = makeStyles(theme => ({
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center'
	},
	avatar: {
		margin: theme.spacing(1),
		backgroundColor: theme.palette.secondary.main
	},
	form: {
		width: '100%',
		marginTop: theme.spacing(3)
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
		borderRadius: 20
	}
}))

const Registration = props => {
	const [error, setError] = useState(null)
	console.log('props:', props)
	const classes = useStyles()

	const handleSubmit = async values => {
		const { first_name, last_name, email, password,type } = values

		try {
			const { data } = await axios.post(
				`https://cors-anywhere.herokuapp.com/http://tallyai.us-east-1.elasticbeanstalk.com/api/auth/register`,
				{
					first_name,
					last_name,
					email,
					password,
					type
				}
			)

			data && props.history.push('/Dashboard')
		} catch (err) {
			setError(err.response.data.errors[0])
		}
	}

	return (
		<Container component='main' maxWidth='xs'>
			<CssBaseline />
			<div className={classes.paper}>
				<Typography component='h1' variant='h5'>
					Sign up for an account
				</Typography>
				{error && (
					<Typography variant='overline' color='error'>
						{error}
					</Typography>
				)}
				<Formik
					initialValues={{
						first_name: '',
						last_name: '',
						email: '',
						password: '',
						confirmedPassword: '',
						type:"tally"
					}}
					onSubmit={(values, actions) => handleSubmit(values)}
					validationSchema={RegistrationSchema}
				>
					{props => {
						const {
							touched,
							errors,
							isSubmitting,
							handleBlur,
							handleChange
						} = props
						return (
							<Form className={classes.form} noValidate>
								<Grid container spacing={2}>
									<Grid item xs={12} sm={6}>
										<TextField
											autoComplete='fname'
											name='first_name'
											variant='outlined'
											required
											fullWidth
											id='first_name'
											label='First Name'
											autoFocus
											onChange={handleChange}
											onBlur={handleBlur}
											error={
												errors.first_name && touched.first_name ? true : false
											}
											helperText={
												errors.first_name &&
												touched.first_name &&
												errors.first_name
											}
										/>
									</Grid>
									<Grid item xs={12} sm={6}>
										<TextField
											variant='outlined'
											required
											fullWidth
											id='last_name'
											label='Last Name'
											name='last_name'
											autoComplete='lname'
											onChange={handleChange}
											onBlur={handleBlur}
											error={
												errors.last_name && touched.last_name ? true : false
											}
											helperText={
												errors.last_name &&
												touched.last_name &&
												errors.last_name
											}
										/>
									</Grid>
									<Grid item xs={12}>
										<TextField
											error={errors.email && touched.email ? true : false}
											variant='outlined'
											required
											fullWidth
											id='email'
											label='Email'
											name='email'
											autoComplete='email'
											onChange={handleChange}
											onBlur={handleBlur}
											helperText={errors.email && touched.email && errors.email}
										/>
									</Grid>
									<Grid item xs={12}>
										<TextField
											variant='outlined'
											required
											fullWidth
											name='password'
											label='Password'
											type='password'
											id='password'
											autoComplete='current-password'
											onChange={handleChange}
											onBlur={handleBlur}
											error={errors.password && touched.password ? true : false}
											helperText={
												errors.password && touched.password && errors.password
											}
										/>
									</Grid>
									<Grid item xs={12}>
										<TextField
											variant='outlined'
											required
											fullWidth
											name='confirmedPassword'
											label='Confirm Password'
											type='password'
											id='confirmedPassword'
											onChange={handleChange}
											onBlur={handleBlur}
											error={
												errors.confirmedPassword && touched.confirmedPassword
													? true
													: false
											}
											helperText={
												errors.confirmedPassword &&
												touched.confirmedPassword &&
												errors.confirmedPassword
											}
										/>
									</Grid>
								</Grid>
								{isSubmitting && <LinearProgress />}
								<Button
									type='submit'
									variant='contained'
									color='primary'
									className={classes.submit}
									fullWidth
									disabled={
										(errors.first_name && touched.first_name) ||
										(errors.last_name && touched.last_name) ||
										(errors.email && touched.email) ||
										(errors.password && touched.password) ||
										(errors.confirmedPassword && touched.confirmedPassword)
											? true
											: false
									}
								>
									Sign Up
								</Button>
								<Grid container justify='center'>
									<Grid item>
										<p>
											Already have an account?{' '}
											<Link
												to='/Login'
												style={{ fontSize: 14, color: '#0000EE' }}
											>
												Sign in
											</Link>
										</p>
									</Grid>
								</Grid>
							</Form>
						)
					}}
				</Formik>
				<GoogleLoginBtn />
			</div>
		</Container>
	)
}

export default Registration
