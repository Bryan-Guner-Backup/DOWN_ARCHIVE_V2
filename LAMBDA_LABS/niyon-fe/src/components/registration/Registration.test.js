import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Registration from './Registration'

describe('<Registration /> component testing', () => {
  let component
  let mockSubmit
  beforeEach(() => {
    mockSubmit = jest.fn()
    component = shallow(<Registration onSubmit={mockSubmit} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should pass snaps shot testing', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render <Registration /> component correctly', () => {
    expect(component.exists()).toBe(true)
  })

  it('should render form correctly with class name formRegister', () => {
    expect(component.find('form').exists()).toBe(true)
    expect(component.find('form').hasClass('formRegister')).toBe(true)
  })

  it('should render the div container correctly', () => {
    expect(component.find('div').at(0).exists()).toBe(true)
  })

  it('should render the input for email correctly', () => {
    expect(component.find('input').at(0).exists()).toBe(true)
  })

  it('should render the input for email correctly', () => {
    expect(component.find('input').at(1).exists()).toBe(true)
  })

  it('should render select element', () => {
    expect(component.find('select').length).toBe(1)
  })

  it('should render Mentor section', () => {
    expect(component.find('select').find('option').at(0).text()).toEqual('Mentor')
  })

  it('should render Mentee section', () => {
    expect(component.find('select').find('option').at(1).text()).toEqual('Mentee')
  })

  it('should render submit button correctly with its text', () => {
    const button = component.find('button')
    expect(button.exists()).toBe(true)
    expect(button.text()).toEqual('Register')
  })
})
