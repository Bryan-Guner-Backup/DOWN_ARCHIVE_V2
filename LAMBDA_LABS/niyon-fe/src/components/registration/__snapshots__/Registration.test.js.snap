// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> component testing should pass snaps shot testing 1`] = `
<div
  className="formWrap"
>
  <Header />
  <form
    className="formRegister"
    onSubmit={[Function]}
  >
    <input
      name="email"
      placeholder="Email"
      type="email"
    />
    <input
      name="password"
      placeholder="Password"
      type="password"
    />
    <input
      name="password2"
      placeholder="Repeat Password"
      type="password"
    />
    <select
      name="user_type"
    >
      <option
        value="Mentor"
      >
        Mentor
      </option>
      <option
        value=" Mentee"
      >
        Mentee
      </option>
    </select>
    <button
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
